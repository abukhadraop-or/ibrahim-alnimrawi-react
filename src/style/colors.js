const colors = {
  black: '#000000',
  darkCyan_limeGreen: '#09733a',
  darkGray: '#999999',
  darkGrayishBlue: '#7d858e',
  gray: '#bbbbbb',
  lightGrayishBlue: '#b8c3e0',
  Mercury: '#E5E5E5',
  moderateLimeGreen: '#5cb85c',
  veryDarkGrayishBlue: '#373a3c',
  veryDarkGrayishBlue2: '#404b59',
  veryDarkGrayishBlue3: '#2F3236',
  veryLightGray_mostlyWhite: '#f3f3f3',
  veryLightGray: '#dddddd',
  white: '#ffffff',
};
export default colors;
