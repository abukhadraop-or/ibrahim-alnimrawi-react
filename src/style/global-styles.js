import { createGlobalStyle } from 'styled-components';

/* istanbul ignore next */
const GlobalStyle = createGlobalStyle`
  body {
    font-family: Serif;
    *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
  padding: 0;
  margin: 0;
  }
`;

export default GlobalStyle;
