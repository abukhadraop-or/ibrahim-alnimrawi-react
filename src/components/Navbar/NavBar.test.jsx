import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import NavBar from 'components/Navbar/Navbar';
import Enzyme, { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { LinksContainer, MenuBar, MenuLink } from './navbar.style';

Enzyme.configure({ adapter: new Adapter() });
describe('<NavBar />', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('MenuBar should call onClick function once ', () => {
    const wrapper = mount(<NavBar />);
    const menuBar = wrapper.find(MenuBar);
    menuBar.simulate('click');
    expect(wrapper.find(LinksContainer).length).toBe(1);
    const menu = wrapper.find(MenuLink).first();
    menu.simulate('click');
    expect(wrapper.find(LinksContainer).length).toBe(0);
  });
});
