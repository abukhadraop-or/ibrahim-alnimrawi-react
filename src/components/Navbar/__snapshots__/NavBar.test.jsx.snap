// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match the snapshot 1`] = `
.c0 {
  color: rgba(0,0,0,0.3);
  display: grid;
  grid-column-gap: 1rem;
  grid-template-columns: 3fr auto auto auto;
  margin: 1rem 7rem;
}

.c2 {
  color: #b8c3e0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:hover {
  color: #404b59;
}

.c3 {
  background-image: url(MenuBar.svg);
  background-size: cover;
  cursor: pointer;
  height: 2.5rem;
  margin-right: 1.5rem;
  width: 2.5rem;
}

.c1 {
  color: #5cb85c;
  font-weight: 1000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (max-width:34.375rem) {
  .c0 {
    grid-template-columns: 3fr auto;
    margin: 1rem 0rem;
  }
}

@media (max-width:34.375rem) {
  .c2 {
    display: none;
  }
}

@media (min-width:34.438rem) {
  .c3 {
    display: none;
  }
}

@media (max-width:34.375rem) {
  .c1 {
    font-size: 1.5rem;
    margin-left: 2.5rem;
  }
}

<div
  className="c0"
>
  <a
    aria-current="page"
    className="c1 active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
     Conduit 
  </a>
  <a
    aria-current={null}
    className="c2"
    href="/Home"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    aria-current={null}
    className="c2"
    href="/SignIn"
    onClick={[Function]}
  >
    Sign in
  </a>
  <a
    aria-current={null}
    className="c2"
    href="/SignUp"
    onClick={[Function]}
  >
    Sign up
  </a>
  <div
    className="c3"
    onClick={[Function]}
  />
</div>
`;
